source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -595,6 +601,33 @@ std::vector<Ssl::PrivateKeyMethodProviderSharedPtr> ContextImpl::getPrivateKeyMe
   return providers;
 }
 
+void ContextImpl::updateCertStats(Stats::Store& store) {
+  // Add CA cert expiration time.
+  std::vector<Ssl::CertificateNameExpiration> certs_expiration_time =
+      cert_validator_->getExpirationOfCerts();
+
+  // Add TLS certs' expiration time.
+  for (const auto& ctx : tls_contexts_) {
+    auto seconds_until_expiration =
+        Utility::getSecondsUntilExpiration(ctx.cert_chain_.get(), time_source_);
+    if (seconds_until_expiration.has_value()) {
+      certs_expiration_time.emplace_back(Ssl::CertificateNameExpiration{
+          ctx.cert_name_, std::chrono::duration<uint64_t>(seconds_until_expiration.value())});
+    }
+  }
+
+  for (const auto& cert : certs_expiration_time) {
+    const std::string& cert_name = cert.name;
+    uint64_t expiration_date = cert.expiration.count();
+    if (cert_stats_map_.find(cert_name) == cert_stats_map_.end()) {
+      cert_stats_map_[cert_name].scope = store.createScope(absl::StrCat("cert.", cert_name, "."));
```

Comment:
You don't need a separate scope for each stat; they should all be in `scope_` which already exists

Reply:
Removed this part of the code since we don't create the stat here anymore.